src/main/java/gov/nasa/cumulus/CNMResponse.java

```diff
@@ -232,6 +232,34 @@ public String buildGeneralError(JsonObject input, String cause) {
         return new Gson().toJson(failureJson);
     }
 
+    /**
+     * Gets the value for the 'dataProcessingType' field, to be used
+     * by the 'buildMessageAttributesHash' method.
+     * <br><br>
+     * Tries to use the 'OriginalCNM > Product > dataProcessingType' from the
+     * 'input > config' section, of the provided Json, but in case of an error,
+     * return a generic error string.
+     *<br><br>
+     * @param input     the input > config section, as a JsonObject
+     * @return          the string to use as the 'dataVersion' value
+     */
+    public String getDataProcessingType(JsonObject input) {
+        String dataProcessingType = null;
+        // Implement a recursive json search function (find "dataProcessingType" else)
+        try {
+            if(input.has("OriginalCNM") &&
+                    input.getAsJsonObject("OriginalCNM").has("product") &&
+                    input.getAsJsonObject("OriginalCNM").getAsJsonObject("product").has("dataProcessingType")) {
+                dataProcessingType = input.getAsJsonObject("product").get("dataProcessingType").getAsString();
+            }
+            return dataProcessingType;
+        } catch (Exception e) {
+            AdapterLogger.LogError(this.className + " handleRequest error:\n" + e.getMessage());
+            AdapterLogger.LogInfo("input content:\n" + input);
+            return "Unknown/Missing";
```

Comment:
Do we want to return "Unknown/Missing"? We should just return null since in buildMessageAttributesHash if dataProcessingType is null we don't add as an attribute.